lib/has_content_id.rb

```diff
@@ -0,0 +1,14 @@
+module HasContentId
+  extend ActiveSupport::Concern
+
+  included do
+    before_validation :generate_content_id, on: :create
+    validates :content_id, presence: true
+  end
+
+private
```

Comment:
The problem with doing it in the block form is that it then gets complex to test. You end up having to build up an 'almost' ActiveRecord object to get the callback to fire. I appreciate that with my approach we end up testing the private method but going this way allows us to test the two 'parts' of the behaviour separately - i) is the behaviour hooked up to the event and ii) does the behaviour do what it is supposed to.